src/celengine/star.cpp

```diff
@@ -945,6 +945,13 @@ StarDetails::setInfoURL(const string& _infoURL)
 }
 
 
+Star::Star(Star &&s) : AstroObject(std::move(s))
+{
+    setPosition(s.getPosition());
+    setAbsoluteMagnitude(s.getAbsoluteMagnitude());
+    setDetails(s.getDetails());
```

Comment:
please use direct access to private members

Reply:
Why You prefer direct access? Using accessor seems to be more stable.

Reply:
outside of the class - yes. but not here. first, this is internal class implementation. second it must set moved object's properties to "null" (nullptr for pointers, etc), you should not duplicate or add validation layers here, just take there, put here and make destructor called on the moved object essentially "nop".

Reply:
Ok, essentially understood.